code/game/objects/structures/xeno.dm

```diff
@@ -117,6 +117,9 @@
 		return FALSE
 
 	if(X.a_intent == INTENT_HARM) //Clear it out on hit; no need to double tap.
+		// No funny spitting and then removing thin spit to get free quick-builds
+		if(src.type == /obj/alien/resin/sticky && (SSmonitor.gamestate == SHUTTERS_CLOSED && CHECK_BITFIELD(SSticker.mode.flags_round_type, MODE_ALLOW_XENO_QUICKBUILD)))
```

Comment:
Never ever use src.ype == . In general, rarely use types, if ever

If some types have a behaviour and other not, it should either use inheritance or vars